spi/src/main/java/org/opensearch/jobscheduler/spi/LockModel.java

```diff
@@ -75,8 +139,26 @@ public LockModel(final LockModel copyLock, long seqNo, long primaryTerm) {
      * @param released boolean flag to indicate if the lock is released
      */
     public LockModel(final LockModel copyLock, final boolean released) {
-        this(copyLock.jobIndexName, copyLock.jobId, copyLock.lockTime, copyLock.lockDurationSeconds,
-            released, copyLock.seqNo, copyLock.primaryTerm);
+        if (copyLock.isResourceLock) {
+            this.lockId = copyLock.lockId;
+            this.resource = copyLock.resource;
+            this.resourceType = copyLock.resourceType;
+            this.isResourceLock = true;
+        }
+        else {
+            this.lockId = copyLock.jobIndexName + LOCK_ID_DELIMITER + copyLock.jobId;
+            this.resource = null;
+            this.resourceType = null;
+            this.isResourceLock = false;
+        }
+        this.jobId = copyLock.jobId;
+        this.jobIndexName = copyLock.jobIndexName;
+        this.lockTime = copyLock.lockTime;
+        this.lockDurationSeconds = copyLock.lockDurationSeconds;
+        this.seqNo = copyLock.seqNo;
+        this.primaryTerm = copyLock.primaryTerm;
+        this.released = released;
+
```

Comment:
same as other comments - call fully parameterized constructor with correct values instead of setting here directly